monitoring/iostats_context.cc

```diff
@@ -10,16 +10,12 @@
 
 namespace rocksdb {
 
-ThreadLocalPtr iostats_context([](void* ptr) {
-    auto* p = static_cast<IOStatsContext*>(ptr);
-    delete p;
-  });
+#ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
+__thread IOStatsContext iostats_context;
+#endif
 
 IOStatsContext* get_iostats_context() {
-  if (iostats_context.Get() == nullptr) {
-    iostats_context.Reset(static_cast<void*>(new IOStatsContext()));
-  }
-  return static_cast<IOStatsContext*>(iostats_context.Get());
+  return &iostats_context;
```

Comment:
I don't get it. Will it even compile if ROCKSDB_SUPPORT_THREAD_LOCAL is not set?